File: cs50x/week 2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // Get input text
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    // Loop through each character
    for (int i = 0; text[i] != '\0'; i++)
    {
        // Count letters with cases
        if (isalpha(text[i]))
        {
            letters++;
        }

        // Count words with hitting spaces
        if (text[i] == ' ')
        {
            words++;
        }

        // Count sentences with hitting punctuation marks
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // Last word with no space after it
    words++;

    // Calculate averages per 100 words
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;

    // Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Rounding the result
    int grade = round(index);

    // Output result
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: cs50x/week 6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Get user input
    text = get_string("Text: ")

    # Initialize counters
    letters = 0
    words = 0
    sentences = 0

    # Iterate through the text to count components
    for char in text:
        # Count letters (A-Z or a-z)
        if char.isalpha():
            letters += 1

        # Count sentence endings (. ! ?)
        elif char in ['.', '!', '?']:
            sentences += 1

    # Count words (assuming words are separated by spaces)
    # split() handles multiple spaces and leading/trailing whitespace effectively
    words = len(text.split())

    # Coleman-Liau formula logic
    # L is average letters per 100 words
    # S is average sentences per 100 words
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Calculate index
    index = 0.0588 * L - 0.296 * S - 15.8

    # Round to nearest integer
    grade = round(index)

    # Output results based on grade thresholds
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


if __name__ == "__main__":
    main()
